src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ImportStatusBauService.cs

```diff
@@ -48,35 +51,62 @@ public class ImportStatusBauService : IImportStatusBauService
+                        .Where(result =>
+                        {
+                            var releaseId = new Guid(result.PartitionKey);
+                            var dataFileName = result.RowKey;
+                            var release =
+                                _contentDbContext.Releases
+                                    .Where(r => r.Id == releaseId)
+                                    .Include(r => r.Publication)
+                                    .FirstOrDefault();
+                            if (release != null)
+                            {
+                                _logger.LogWarning("No ContentDB Releases table row found!\n" +
+                                                   $"releaseId: \"{releaseId}\"");
+                            }
+
+                            var releaseFileRef = _contentDbContext.ReleaseFileReferences.FirstOrDefault(r =>
+                                r.ReleaseId == releaseId &&
+                                r.Filename == dataFileName &&
+                                r.ReleaseFileType == ReleaseFileTypes.Data);
+                            if (releaseFileRef != null)
+                            {
+                                _logger.LogWarning("No ContentDB ReleaseFileReferences table row found for data file!\n" +
+                                                   $"releaseId: \"{releaseId}\"\n" +
+                                                   $"dataFileName: \"{dataFileName}\"");
+                            }
+                            return release != null && releaseFileRef != null;
+                        })
                         .Select(async result =>
-                    {
-                        var releaseId = new Guid(result.PartitionKey);
-                        var dataFileName = result.RowKey;
-                        var release =
-                            await _contentDbContext.Releases
-                                .Where(r => r.Id == releaseId)
-                                .Include(r => r.Publication)
-                                .FirstAsync();
+                        {
+                            var releaseId = new Guid(result.PartitionKey);
+                            var dataFileName = result.RowKey;
+                            var release =
+                                await _contentDbContext.Releases
+                                    .Where(r => r.Id == releaseId)
+                                    .Include(r => r.Publication)
+                                    .FirstAsync();
 
-                        var releaseFileRef = await _contentDbContext.ReleaseFileReferences.FirstAsync(r =>
-                            r.ReleaseId == releaseId &&
-                            r.Filename == dataFileName &&
-                            r.ReleaseFileType == ReleaseFileTypes.Data);
+                            var releaseFileRef = await _contentDbContext.ReleaseFileReferences.FirstAsync(r =>
```

Comment:
Would just get the `ReleaseFileReferences` and use an `Include` and `ThenInclude` to get the associated Release and Publication.